cmd/oras/root/manifest/index/update.go

```diff
@@ -181,6 +181,7 @@ func addManifests(ctx context.Context, manifests []ocispec.Descriptor, target or
 			return nil, fmt.Errorf("%s is not a manifest", manifestRef)
 		}
 		printUpdateStatus(status.IndexPromptFetched, manifestRef, string(desc.Digest), opts.Printer)
+		desc = descriptor.Plain(desc)
```

Comment:
Is it better move the plain descriptor conversion and manifest media type comparison into getPlatform?

Reply:
But `getPlatform` is only reached when the descriptor is image manifest (see the if statement in the next line). We need to do plain descriptor for all kinds of manifests.

Reply:
What I mean is add a new function to wrap the descriptor processing, and reuse it in both update and create.

```golang
func enrichDescriptor(ctx context.Context, target oras.ReadOnlyTarget, desc ocispec.Descriptor) (ocispec.Descriptor, error) {
	desc = descriptor.Plain(desc)
	if descriptor.IsImageManifest(desc) {
		if desc.Platform, err = getPlatform(ctx, target, content); err != nil {
			return nil, err
		}
	}
	return desc, nil
}
```

Reply:
changed accordingly.